src/products/repository/productCategories.repository.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import ProductCategory from '../domain/productCategory.domain';
+
+@Injectable()
+export default class ProductCategoriesRepository {
+  constructor(
+    @InjectRepository(ProductCategory)
+    private productCategoriesRepository: Repository<ProductCategory>,
+  ) {}
+
+  findAll(): Promise<ProductCategory[]> {
+    return this.productCategoriesRepository.find();
+  }
+
+  findOne(id: string): Promise<ProductCategory | null> {
+    return this.productCategoriesRepository.findOneBy({ id });
+  }
+
+  create(order: ProductCategory): Promise<ProductCategory> {
+    return this.productCategoriesRepository.save(order);
+  }
+
+  update(order: ProductCategory): Promise<ProductCategory> {
+    return this.productCategoriesRepository.save(order);
+  }
```

Comment:
You can create only one method (`save`) if `create` and `update` are identical.

Same for other files.

Reply:
done, did it in other repositories too